lang/java/reef-common/src/main/java/org/apache/reef/driver/parameters/DriverRestartHandler.java

```diff
@@ -23,10 +23,12 @@
 import org.apache.reef.wake.EventHandler;
 import org.apache.reef.wake.time.event.StartTime;
 
+import java.util.Set;
+
 /**
  * The StartTime event is routed to this EventHandler if there is a restart, instead of to DriverStartHandler.
  */
 @NamedParameter(doc = "The StartTime event is routed to this EventHandler if there is a restart, " +
     "instead of to DriverStartHandler.")
-public final class DriverRestartHandler implements Name<EventHandler<StartTime>> {
+public final class DriverRestartHandler implements Name<Set<EventHandler<StartTime>>> {
```

Comment:
As it is presumably OK to break compatibility in this area of the REEF API, we should also consider changing the event type here to `DriverRestarted` or such. That way, we can add additional information to it over time (e.g. the attempt). That would be a good candidate for a separate JIRA/PR, though.